### PR TITLE
Update UI for Payment Task Screen

### LINKED ISSUE #6683: Payments task: improving how we surface WCPayments
**Description:** the WooCommerce Payments card in the Payments task is not performing as expected. We're going to implement some changes in the card to provide additional information and reinforce the value prop.

Choose payment methods

Completed setup task list

**Acceptance criteria:** 

- [ ] Implement the new layout + copy
- [ ] "Learn more" should link to: https://woocommerce.com/payments/
- [ ] Clicking "Get started" will redirect users to the WCPayments KYC flow (as we have right now)
- [ ] All other payment methods are located in a separate card titled "Additional payment methods".
- [ ] Update the WCPayments card to the layout displayed above when the user completes the setup
- [ ] Update the WCPayments card with the following if the user visits the payments task after installing the extension but without completing the setup:
Switch the "recommended" pill to "Setup required"
Update the CTA copy to "Finish setup"
([mockup](https://github.com/woocommerce/woocommerce-admin/issues/6683#issuecomment-811173292) in this comment)
- [ ] Redirect to the WCPayments dashboard when the user clicks "Manage"

**Additional event tracking:**
`wcadmin_tasklist_payment_learn_more`: triggered when the user clicks in the WCPayments learn more link
`wcadmin_tasklist_payment_manage`: triggered when the user clicks in the WCPayments manage link

**Additional links:**
Project thread: paJDYF-1eJ-p2
[Design mockup](https://www.figma.com/file/7QOkvxzz1ylS3QBS4zqW8R/Payments-experiments?node-id=1424%3A327)

### PR DESCRIPTION
**Pull request by joelclimbsthings**
Fixes #6683 

Implementing new UI for payment task. 

**Outstanding questions**
- I'm a little uncertain of how to handle the image assets, namely the WCPay logo & credit card images. After asking around I was pointed towards [these assets already in the woocommerce repo](https://github.com/woocommerce/woocommerce/tree/23710744c01ded649d6a94a4eaea8745e543159f/assets/images/icons/credit-cards), but they're not all there, those that are have a different appearance, and I'm not sure of the pattern for referencing assets in another plugin.
 Right now I have them all in their own SVG files, although that of course drives up the plugin size a bit. I can also convert the CC icons to PNG files if desired, since those would be significantly lighter in that case (if we don't intent to use these in larger sizes elsewhere).

**WCPay not enabled**
image

**WCPay enabled but not configured**
image

**WCPay enabled & configured**
image

**Mobile view**
image

**Acceptance criteria:** 

- [x] Implement the new layout + copy
- [x] "Learn more" should link to: https://woocommerce.com/payments/
- [x] Clicking "Get started" will redirect users to the WCPayments KYC flow (as we have right now)
- [x] All other payment methods are located in a separate card titled "Additional payment methods".
- [x] Update the WCPayments card to the layout displayed above when the user completes the setup
- [x] Update the WCPayments card with the following if the user visits the payments task after installing the extension but without completing the setup:
Switch the "recommended" pill to "Setup required"
Update the CTA copy to "Finish setup"
([mockup](https://github.com/woocommerce/woocommerce-admin/issues/6683#issuecomment-811173292) in this comment)
- [x] Redirect to the WCPayments dashboard when the user clicks "Manage"

### Accessibility

-   [x] I've tested using only a keyboard (no mouse)
-   [ ] I've tested using a screen reader

### Detailed test instructions:

-  Create fresh site and install plugin(s).
- Navigate to Homescreen & click "Choose payment methods" task.
- You should see the UI from the initial screenshot, with the WCPay Card including the "recommended" pill and "Get started" button.
- Ensure "Learn more" link and "Get started" buttons work as per the acceptance criteria. 
- Confirm `wcadmin_tasklist_payment_learn_more` tracks event is fired while clicking "Learn more."
- Check mobile view (last screenshot).
- Enable WCPay, but don't configure (I do this by overriding boolean values in `methods.js`
- View screen again, which should now appear like the second screenshot, including "Setup required" icon & "Finish setup" button.
- Configure WCPay (again, I do this through overriding boolean in `methods.js`).
- Now you should see WCPay appear in "Enabled payment methods" card, as per the third screenshot.
- Test "Manage" button for WCPay, and ensure `wcadmin_tasklist_payment_manage` tracks event is fired.
- Check mobile view again.
- Quick smoke test that includes the rest of the functionality on this page, including "Set up" buttons for other payment methods.

### CODE REVIEW

**Review by psealock — COMMENTED** [review]
Are there contractual obligations to keeping the "Recommended" banner and dark blue call to action button on Stripe? Otherwise we have two "Recommended" and it takes away from WC Pay. cc @pmcpinto 

Screen Shot 2021-04-12 at 4 27 32 PM

**Comment by pmcpinto** [comment]
> Are there contractual obligations to keeping the "Recommended" banner and dark blue call to action button on Stripe? Otherwise we have two "Recommended" and it takes away from WC Pay. cc @pmcpinto

We're not required to have the "recommended" ribbon/pill and the blue CTA on Stripe when WCPayments is promoted. Thanks

**Comment by joelclimbsthings** [comment]
Good observation @psealock . I [was wondering the same](https://github.com/woocommerce/woocommerce-admin/issues/6683#issuecomment-816885168), as was Elizabeth. 

Just reverted the relevant logic to only have recommended gateways when WCPay is _not_ promoted.

**Comment by psealock** [comment]
>I'm a little uncertain of how to handle the image assets, namely the WCPay logo & credit card images. After asking around I was pointed towards these assets already in the woocommerce repo, but they're not all there, those that are have a different appearance, and I'm not sure of the pattern for referencing assets in another plugin.

For those that are a different appearance than the designs I think thats ok. We could also change the images in Core and keep the filename the same.

The ones that aren't there should be added to Core. We'll need to also add them to WC Admin for backwards compatibility reasons.

The WC Pay logo could be added to our repo, but maybe there is a case for it being in Core. We can open a pull request and have the discussion there.

As for referencing another plugin's assets, I'll poke around. I believe we've done something like this in the past.

File: client/task-list/tasks/payments/components/WCPayCard/WCPayCard.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import { recordEvent } from '@woocommerce/tracks';
+
+/**
+ * Internal dependencies
+ */
+import './WCPayCard.scss';
+import WCPayAcceptedMethods from '../WCPayAcceptedMethods';
+import WCPayLogo from '../../images/wcpay-logo';
+
+export default ( { method, handleClick, isBusy } ) => {
```
Comment (joshuatf) [inline client/task-list/tasks/payments/components/WCPayCard/WCPayCard.js:25]: Maybe we could just pass `isEnabled` here?  Do we need the whole `method`?

File: client/task-list/tasks/payments/index.js [hunk 2]
```diff
@@ -254,6 +250,21 @@ class Payments extends Component {
 				</div>
 			);
 		}
+
+		if ( method.key === 'wcpay' ) {
```
Comment (joshuatf) [inline client/task-list/tasks/payments/index.js:254]: Heads up that I updated the keys in https://github.com/woocommerce/woocommerce-admin/pull/6786 to match the slugs used for payment settings pages.  This one will be `woocommerce_payments`.

The below section is probably also part of the `Action` component.

**Review by joshuatf — COMMENTED** [review]
Thanks for putting this together, Joel 🎉 

Added some comments if we merge https://github.com/woocommerce/woocommerce-admin/pull/6786 first, but let me know if you feel it will be easier the other way around.

File: client/task-list/tasks/payments/components/WCPayCard/WCPayCard.js (hunk 1 above)
Comment (joelclimbsthings) [reply client/task-list/tasks/payments/components/WCPayCard/WCPayCard.js]: That _would_ have been the case, although we're now using a number of properties to pass to the `<Action />` component.

**Comment by joelclimbsthings** [comment]
> The WC Pay logo could be added to our repo, but maybe there is a case for it being in Core. We can open a pull request and have the discussion there.

It sounds like regardless all of the images should be added to `wc-admin` for now, with the possibility of switching over to woocommerce assets in the future if we can get those added @psealock ?

Thanks @joshuatf & @psealock for the review. I've just finished a fairly heavy rebase and pushed the results, so I'll need another run through the testing instructions to verify.

File: client/task-list/tasks/payments/components/Action.js [hunk 3]
```diff
@@ -75,10 +76,12 @@ export const Action = ( {
...
 					className={ classes }
 					isSecondary
-					href={ getAdminLink(
-						'admin.php?page=wc-settings&tab=checkout&section=' +
-							methodKey
-					) }
+					href={ manageUrl }
+					onClick={
+						methodKey === 'wcpay'
+							? () => recordEvent( 'tasklist_payment_manage' )
```
Comment (joshuatf) [inline client/task-list/tasks/payments/components/Action.js:82]: Seems weird we're only recording this for `wcpay`.  Should we be recording this for all methods with a prop for the method key?  cc @pmcpinto

File: client/task-list/tasks/payments/components/PaymentMethodList/PaymentMethodList.scss [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+		}
+
+		.woocommerce-task-payment__description {
+			order: 3;
+			padding: $gap-large 0 0 $gap-large;
+		}
+
+		.woocommerce-task-payment__footer {
+			flex-basis: 50%;
+			align-self: flex-end;
```
Comment (joshuatf) [inline client/task-list/tasks/payments/components/PaymentMethodList/PaymentMethodList.scss:101]: I feel like on smaller screens this should be flex-start or center.  What do you think?

Current:
Screen Shot 2021-04-15 at 12 11 14 PM


Start:
Screen Shot 2021-04-15 at 12 11 32 PM

**Review by joshuatf — APPROVED** [review]
Great work and nice job tackling that complicated rebase, Joel 💯 

Added a few minor comments/questions, but pre-approving as nothing that's a blocker.

File: client/task-list/tasks/payments/components/RecommendedRibbon.js [hunk 5]
```diff
@@ -3,7 +3,7 @@
  */
 import { __ } from '@wordpress/i18n';
 
-const localPartners = [ 'mercadopago' ];
+const localPartners = [ 'woo-mercado-pago-basic' ];
```
[the remark this answers is not in the document]
Comment (joelclimbsthings) [reply client/task-list/tasks/payments/components/RecommendedRibbon.js:6]: Nice catch, had some funny business with the re-rebase, but thought I got all these.

File: client/task-list/tasks/payments/components/Action.js (hunk 3 above)
Comment (joelclimbsthings) [reply client/task-list/tasks/payments/components/Action.js:82]: Good question, it sounded wcpay specific to me from the requirements, but I'll let Pedro weigh in.

File: client/task-list/tasks/payments/components/PaymentMethodList/PaymentMethodList.scss (hunk 4 above)
Comment (joelclimbsthings) [reply client/task-list/tasks/payments/components/PaymentMethodList/PaymentMethodList.scss]: I think I like it. Done in bfbec15cbfc2d27f5732c60879b7d02145c3e720

File: client/task-list/tasks/payments/components/Action.js (hunk 3 above)
Comment (joelclimbsthings) [reply client/task-list/tasks/payments/components/Action.js:82]: I've merged this PR, but will address this with a follow-up I currently have going depending on the response.

**Pull request merged**